### PR TITLE
Add deprecated-editor-group flag to existing call to get editor details

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

* Adds a flag to indicate that the user has been added to the deprecated editor group

Related diff D44256-code

#### Testing instructions

* Run this PR locally logging in with user that doesn't match Diff D44256's requirements to be in the deprecation group
* Edit a post
* Inspect the Redux state and confirm that under `currentUser` there is a property `inEditorDeprecationGroup` set to false.
* Login with a user that does match D44256's requirements to be in the deprecation group or sandbox API and fudge that value in the API response to be truthy
* Switch back to editing a post
* Confirm the state `currentUser.inEditorDeprecationGroup` value is now set to true.

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/gutenberg/index.js [hunk 1]
```diff
@@ -31,7 +31,12 @@ const fetchGutenbergOptInData = ( action ) =>
 
 const setGutenbergOptInData = (
 	{ siteId },
-	{ editor_web: editor, opt_in: optIn, opt_out: optOut }
+	{
+		editor_web: editor,
+		opt_in: optIn,
+		opt_out: optOut,
+		in_editor_deprecation_group: inEditorDeprecationGroup,
```
Comment (glendaviesnz) [inline client/state/data-layer/wpcom/sites/gutenberg/index.js:38]: This needs plumbing through to reducer and a selector adding in order to retrieve it

**Comment by aaronrobertshaw** [comment]
@glendaviesnz I've had a go at plumbing all this through Calypso. I might have been a little trigger happy and pushed a couple of changes in a commit I didn't mean to. But they actually use the new flag so give us a way of testing. I'll leave them there for now and will go through it with you tomorrow.

**Comment by glendaviesnz** [comment]
@aaronrobertshaw this tests good for me, but I think it would be good to get sign-off from another member of the team that hasn't been closely involved in working through this. Do you want to try a rebase and push to see if that sorts the failed tests?

**Comment by scruffian** [comment]
When I switch back to the Classic Editor, I find that other links in Calypso still take me to the Calypso editor. I'll work on updating those...

**Comment by scruffian** [comment]
I added a commit to update the edit post links to go to the wp-admin Classic Editor. I also wonder if we should add a redirect from `/post` in this case too, just incase there are other links that aren't using `getEditorUrl`.

In some ways I'd prefer we always pointed user at the same URL for editing posts (/post) and then just redirect based on their preferences. What do you think? It would keep all the logic in one place...

**Comment by scruffian** [comment]
> I also wonder if we should add a redirect from /post in this case too, just in case there are other links that aren't using getEditorUrl.

I added a commit for this.

**Comment by aaronrobertshaw** [comment]
>>I also wonder if we should add a redirect from /post in this case too, just in case there are other links that aren't using getEditorUrl.
>
>I added a commit for this.

@scruffian I might have missed where that commit was, I can't see it. 

I've added something just to redirect anyone visiting the /post urls etc. If there is a better approach to achieving the redirects let me know and I'll update it. No other changes to consolidate logic to a single location have been made yet. I figure once the approach is confirmed ok or updated, that consolidation can be done in a separate PR.

@glendaviesnz I've rebased this PR and re-pushed, I'll check on whether it fixes the failing tests tomorrow.

File: client/post-editor/index.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ export default function () {
 		'/post/:site?/:post?',
 		siteSelection,
 		controller.gutenberg,
+		controller.classic,
```
Comment (scruffian) [inline client/post-editor/index.js:21]: Another option would be to use the `maybeRedirect` function.

**Comment by scruffian** [comment]
> @scruffian I might have missed where that commit was, I can't see it.

You're right I didn't push it, sorry :(

File: client/post-editor/index.js (hunk 2 above)
Comment (aaronrobertshaw) [reply client/post-editor/index.js:21]: I've tried updating the `maybeRedirect` function in its current place though that results in a flash of the editor being loaded before the actual redirect to WP admin takes affect. 

I could go about pulling it out of the `startEditing` function and moving it earlier in the middleware chain. Is it worth doing that now? Is it something that could be done if/when we move the redirect logic from all the other links into one location?

Comment (scruffian) [reply client/post-editor/index.js:21]: I think the current solution is good 👍

**Comment by glendaviesnz** [comment]
I have retested this and all works as advertised - 🚢

**Pull request merged**